### PR TITLE
Add support for chrono::DateTime<T>

### PR DESCRIPTION
**Pull request by emwalker**
Draft PR to add support for converting to and from `chrono::DateTime<Utc>` and `chrono::DateTime<FixedOffset>` timestamps.  [Related discussion](https://github.com/matsadler/magnus/issues/113).  I'm out of my depth here both with respect to the Ruby C API and with respect to usage of `chrono`, so additional diligence might be needed.

I am not attached to this PR at all.  With support for nanosecond precision recently added (still to be tested on our side), it is possible that we will not need `chrono`.

### CODE REVIEW

File: src/time.rs [hunk 1]
```diff
@@ -253,6 +253,30 @@ impl IntoValue for SystemTime {
     }
 }
 
+#[cfg(feature = "chrono")]
```
Comment (matsadler) [inline src/time.rs:256]: If you add `#[cfg_attr(docsrs, doc(cfg(feature = "chrono")))]` after each use of `#[cfg(feature = "chrono")]`, like:

``` rust
#[cfg(feature = "chrono")]
#[cfg_attr(docsrs, doc(cfg(feature = "chrono")))]
impl ...
```

then rustdoc (well, at least with with config that docs.rs uses, run `env RUSTDOCFLAGS='--cfg docsrs' cargo +nightly doc --all-features` to see it locally) will show a little "Available on crate feature chrono only." flag next to this item in the docs.

**Review by matsadler — APPROVED** [review]
This looks good, I'd be happy to merge this as it currently is.

I'd add an entry in the [changelog](https://github.com/matsadler/magnus/blob/main/CHANGELOG.md?plain=1), and a mention here in the readme: 
https://github.com/matsadler/magnus/blob/4109aec2c754a11399feaa8f1f133f3c3622ca15/README.md#L332

similar to how `bytes` get a footnote calling out that it requires a feature:
https://github.com/matsadler/magnus/blob/4109aec2c754a11399feaa8f1f133f3c3622ca15/README.md#L322
https://github.com/matsadler/magnus/blob/4109aec2c754a11399feaa8f1f133f3c3622ca15/README.md#L342

But I'm happy to make those changes myself after merging.

**Comment by emwalker** [comment]
> similar to how bytes get a footnote calling out that it requires a feature:

Given how the table is already a pretty cramped, it seemed like four `****` might be a bit much.  I took the liberty of converting the footnotes to [GitHub Markdown's footnotes](https://github.blog/changelog/2021-09-30-footnotes-now-supported-in-markdown-fields/).  Let me know if you would like me to revert to the four asterisks.

2024-07-22_16-45-footnotes

Unfortunately, the footnotes appear at the bottom of the page instead of the table:

2024-07-22_16-46-footnotes

**Comment by matsadler** [comment]
> it seemed like four `****` might be a bit much. I took the liberty of converting the footnotes to [GitHub Markdown's footnotes](https://github.blog/changelog/2021-09-30-footnotes-now-supported-in-markdown-fields/). 
> ...
> Unfortunately, the footnotes appear at the bottom of the page...

You're right, `****` is a bit much. Probably the `***` that was already there was a bit much.

I didn't know about markdown footnotes, that's neat. The readme is also rendered on crates.io, but I checked and that supports footnotes too.

However I think the way it adds them to the bottom of the page is a dealbreaker for me, especially with such a long page.

Maybe we could use *, †, ‡, §, ||, and ¶ instead of ever more asterisks?

**Comment by emwalker** [comment]
Updated.

In typesetting, I think the symbols would be used in order of *, †, ‡, §, ||, and ¶ as footnote references are found in the table, and then the footnotes themselves would be sorted in that order, in contrast to what I just did.  The symbols would also be superscripts, both in the references and in the footnotes.  But I didn't want to get fancy.  I'm sure you'll update it to something nice.

**Comment by matsadler** [comment]
This is great, thanks!

**Pull request merged**